### ISSUE TITLE
How can I change the y label for a LayerChart?

### ISSUE THREAD

**Issue by trllrgn** [altair-viz/altair#1465]
I am trying to create two different charts, and then combine them using chart = chart1 + chart2. When I do this, the output is a LayerChart, and I get a list of the variable names as the y axis label, e.g. "Varname1, Varname2". I want to rename the y axis label such that it just says "Data Count". I tried to use chart.encode(y="Data Count") but this doesn't do anything.

How can the y label of a LayerChart be modified?

**Comment by jakevdp** [comment]
You can adjust the ``title`` property of the y encoding; for example:

```python
chart.encode(
    y=alt.Y('column_name', title='The Title)
)
```

**Comment by trllrgn** [comment]
Is this supposed to work on the LayerChart level or on the level of Chart objects? I am not sure what to put in place of 'column_name'. Can you please have a look at the dummy code below and let me know how to change it? The label on the y axis is "b,c", and I want to replace that with a text of my choosing.

```
import altair as alt
import pandas as pd

def create_chart(source,var):

    chart = alt.Chart(source).mark_bar(opacity=0.8).encode(
        x='a',
        y=var+':Q'
    )
    return chart

source1 = pd.DataFrame({
    'a': ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I'],
    'b': [28, 55, 43, 91, 81, 53, 19, 87, 52]
})
source2 = pd.DataFrame({
    'a': ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I'],
    'c': [20, 66, 38, 191, 90, 20, 8, 98, 34]
})
    
chart1 = create_chart(source1,'b')
chart2 = create_chart(source2,'c')

chart = chart1+chart2
display(chart)
```

**Comment by jakevdp** [comment]
Your column name in this case is `var + ':Q'`. ``y=var + ':Q'`` is a shorthand for ``y=alt.Y(var, type='quantitative')``.

So to set the axis title, you can change your ``create_chart`` function to the following:
```python
def create_chart(source,var):
    chart = alt.Chart(source).mark_bar(opacity=0.8).encode(
        x='a',
        y=alt.Y(var, type='quantitative', title='Whatever Title You Want')
    )
    return chart
```
![visualization (14)](https://user-images.githubusercontent.com/781659/57083242-0fb25c00-6cad-11e9-9091-fd512e895adf.png)

**Comment by trllrgn** [comment]
Thanks for the suggestion, it partially solves my problem. 

What I would like to have is to display the variable names when only a single chart is plotted. E.g. if it's a plot of variable X, I want the title to say "Data Count X" and when it's only variable Y then the title should say "Data Count Y". However, when multiple charts are plotted in a single LayerChart, then instead of saying "Data Count X, Data Count Y" I just want to replace it with "Data Count".

So basically, individual charts are created with their own titles, and I want to rename the y axis only when multiple variables are combined into a single chart. But at that point, the object is no longer a Chart but a LayerChart instead, and I do not know how to access the y title property of it without changing the underlying chart1 and chart2. 

So can the y label of a LayerChart be set explicitly without know what are the underlying Charts?

**Comment by jakevdp** [comment]
It's not possible to create a single chart whose axis title will change depending on whether or not it appears in a layer. You could adjust the titles after the fact by doing something like this:
```
chart = chart1+chart2
chart.layer[0].encoding.y.title = 'Data Count'
```

**Comment by trllrgn** [comment]
Thanks a lot, this works for me.

Just out of curiosity, the LayerChart object does not have its own title then, it always builds on the titles of the underlying Charts, is that correct? If so, does that mean that all the underlying Charts' titles need to be changed to set the LayerChart's y label?

Thanks again for your help.

**Comment by jakevdp** [comment]
Sort of: the title is always tied to the encoding. If the layer chart itself has an encoding, then the title from it will be used. Otherwise, the titles of the individual layers will be used.

Here's an example of an encoding specified on the layer chart, rather than in the individual subcharts:
```python
import numpy as np
import pandas as pd

df = pd.DataFrame({
    'x': np.arange(100),
    'y1': np.random.randn(100).cumsum(),
    'y2': np.random.randn(100).cumsum(),
})

alt.layer(
    alt.Chart(df).mark_line().encode(y='y1'),
    alt.Chart(df).mark_line().encode(y='y2')
).encode(
    x=alt.X('x:Q', title='The X Title')
)
```

**Comment by jakevdp** [comment]
Also, if only one layer specifies a title and others don't, the specified title will be used.

If multiple layers specify multiple different titles, they will be combined.

**Comment by trllrgn** [comment]
That is great to knows, thanks for the information.
